airflow-core/src/airflow/models/dagbag.py

```diff
@@ -685,12 +685,22 @@ class DagPriorityParsingRequest(Base):
     # Note: Do not depend on fileloc pointing to a file; in the case of a
     # packaged DAG, it will point to the subpath of the DAG within the
     # associated zip.
-    relative_fileloc = Column(String(2000), nullable=False)
+    relative_fileloc = Column(String(2000), nullable=True)
 
-    def __init__(self, bundle_name: str, relative_fileloc: str) -> None:
+    def __init__(self, bundle_name: str, relative_fileloc: str = "") -> None:
```

Comment:
```suggestion
    def __init__(self, bundle_name: str, relative_fileloc: str | None = None) -> None:
```

If we are allowing nullable, lets actually use nulls then - otherwise we don't really need to alter the table in the first place.